src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -97,9 +97,15 @@ public function process(ContainerBuilder $container)
 
             // We must assume that the class value has been correctly filled, even if the service is created by a factory
             $class = $container->getParameterBag()->resolveValue($def->getClass());
-
             $interface = 'Symfony\Component\EventDispatcher\EventSubscriberInterface';
+
             if (!is_subclass_of($class, $interface)) {
+                if (!class_exists($class, true)) {
+                    throw new \RuntimeException(
+                        sprintf('Class "%s" from definition of service "%s" can not be found. Check existence of file and correct naming', $class, $id)
```

Comment:
Here is my [motivation](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Debug/DebugClassLoader.php#L223). If file will exists and has correct name — will be raised that detailed `RuntimeException`.